.github/workflows/pull-gitleaks.yml

```diff
@@ -1,8 +1,9 @@
 name: pull-gitleaks
 on:
-  pull_request:
+  pull_request_target:
```

Comment:
Additionally with target you need to adjust the checkout to https://github.com/kyma-project/test-infra/pull/10153/files#diff-cb4389f8b525d78c165731a8d28493962902fc9617910342d239fb01093d9081R17 or https://github.com/kyma-project/test-infra/blob/main/.github/workflows/pull-plan-prod-terraform.yaml#L30